pkg/controller/ipam/antrea_ipam_controller.go

```diff
@@ -113,20 +127,32 @@ func NewAntreaIPAMController(crdClient versioned.Interface,
 	klog.V(2).InfoS("Subscribing for StatefulSet notifications", "controller", controllerName)
 	statefulSetInformer.Informer().AddEventHandler(
 		cache.ResourceEventHandlerFuncs{
+			AddFunc:    c.enqueueStatefulSetCreateEvent,
 			DeleteFunc: c.enqueueStatefulSetDeleteEvent,
 		},
 	)
 
 	return c
 }
 
+// Enqueue the StatefulSet create notification to be processed by the worker
+func (c *AntreaIPAMController) enqueueStatefulSetCreateEvent(obj interface{}) {
+	ss := obj.(*appsv1.StatefulSet)
+	klog.V(2).InfoS("Create notification", "Namespace", ss.Namespace, "StatefulSet", ss.Name)
+
+	key := k8s.NamespacedName(ss.Namespace, ss.Name)
+
+	c.statefulSetMap.Store(key, ss)
+	c.statefulSetQueue.Add(addEventIndication + key)
+}
+
 // Enqueue the StatefulSet delete notification to be processed by the worker
 func (c *AntreaIPAMController) enqueueStatefulSetDeleteEvent(obj interface{}) {
 	ss := obj.(*appsv1.StatefulSet)
-	klog.V(2).InfoS("Delete notification", "StatefulSet", ss.Name)
+	klog.V(2).InfoS("Delete notification", "Namespace", ss.Namespace, "StatefulSet", ss.Name)
 
 	key := k8s.NamespacedName(ss.Namespace, ss.Name)
-	c.statefulSetCleanupQueue.Add(key)
+	c.statefulSetQueue.Add(delEventIndication + key)
```

Comment:
When we delete ss from statefulSetMap?

Reply:
I feel the current implementation is not right. It cannot handle recreation and annotation change. Probably we should consider the typical controller pattern that I commented earlier:

> I feel we should store/look up previous allocation state in some way, and decide what to do based on the latest ss (get from the client cache) and the stored allocation state. Then recreation can be handled too.